shell/platform/darwin/ios/framework/Source/FlutterKeyboardManagerTest.mm

```diff
@@ -68,50 +31,20 @@ typedef void (^KeyCallbackSetter)(FlutterUIPressProxy* press, FlutterAsyncKeyCal
 
 }  // namespace
 
+// These tests were designed to run on iOS 13.4 or later.
+API_AVAILABLE(ios(13.4))
 @interface FlutterKeyboardManagerTest : XCTestCase
-@property(nonatomic, strong) id mockEngine;
-- (FlutterViewController*)mockOwnerWithPressesBeginOnlyNext API_AVAILABLE(ios(13.4));
 @end
 
 @implementation FlutterKeyboardManagerTest
 
-- (void)setUp {
-  // All of these tests were designed to run on iOS 13.4 or later.
-  if (@available(iOS 13.4, *)) {
-  } else {
-    XCTSkip(@"Required API not present for test.");
-  }
-
-  [super setUp];
-  self.mockEngine = OCMClassMock([FlutterEngine class]);
-}
-
-- (void)tearDown {
-  // We stop mocking here to avoid retain cycles that stop
-  // FlutterViewControllers from deallocing.
-  [self.mockEngine stopMocking];
-  self.mockEngine = nil;
-  [super tearDown];
-}
-
-- (id)checkKeyDownEvent:(UIKeyboardHIDUsage)keyCode API_AVAILABLE(ios(13.4)) {
-  return [OCMArg checkWithBlock:^BOOL(id value) {
-    if (![value isKindOfClass:[FlutterUIPressProxy class]]) {
-      return NO;
-    }
-    FlutterUIPressProxy* press = value;
-    return press.key.keyCode == keyCode;
-  }];
-}
-
-- (id<FlutterKeyPrimaryResponder>)mockPrimaryResponder:(KeyCallbackSetter)callbackSetter
-    API_AVAILABLE(ios(13.4)) {
+- (id<FlutterKeyPrimaryResponder>)mockPrimaryResponder:(KeyCallbackSetter)callbackSetter {
   id<FlutterKeyPrimaryResponder> mock =
       OCMStrictProtocolMock(@protocol(FlutterKeyPrimaryResponder));
   OCMStub([mock handlePress:[OCMArg any] callback:[OCMArg any]])
       .andDo((^(NSInvocation* invocation) {
-        FlutterUIPressProxy* press;
-        FlutterAsyncKeyCallback callback;
+        __unsafe_unretained FlutterUIPressProxy* press;
+        __unsafe_unretained FlutterAsyncKeyCallback callback;
```

Comment:
This seems scary. Why are we passing unsafe unretained pointers to an async block? It seems like this code was probably just wrong before?

Reply:
Given that the next thing we do is pass the arguments into an async block, I think we are missing a critical intermediate step:
> You can then copy this into another form of storage, like a strongly-held property.

I know this is pre-existing, but it seems like a very likely source of flaky test crashing, so we should fix it while we are here. We can rename these something like `pressUnsafe` and `callbackUnsafe` and then do

```objc
FlutterUIPressProxy* press = pressUnsafe;
FlutterAsyncKeyCallback callback = callbackUnsafe;
```

That way we won't be trying to run an async block with pointers into memory owned by an invocation that's almost certainly out of scope. I'm pretty surprised that the current code is ever working.